static/src/js/providers/TourContextProvider/TourContextProvider.jsx

```diff
@@ -11,13 +11,11 @@ const { disableSiteTour } = getApplicationConfig()
 const isSiteTourEnabled = disableSiteTour === 'false'
 
 const TourContextProvider = ({ children }) => {
-  const [runTour, setRunTour] = useState(false)
+  const [runTour, setRunTour] = useState(0)
   useEffect(() => {
     const isLocalhost = window.location.hostname === 'localhost'
     const overrideLocalhost = window.overrideLocalhost === true
-    const hasUserDisabledTour = localStorage.getItem('dontShowTour') === 'true'
-    const shouldShowTour = isSiteTourEnabled && !hasUserDisabledTour
-      && (!isLocalhost || overrideLocalhost)
+    const shouldShowTour = isSiteTourEnabled && (!isLocalhost || overrideLocalhost) ? 1 : 0
```

Comment:
I don't like the magic number even if its a bool I don't understand what this change is for I'd prefer to just move the logic for determining if we are going to showTour into its own function that just returns true or false

Reply:
I don't love this solution myself, but the reason I went with 1 and 0 here, is because the showing of the tour is contingent on the user preference, but we can't access the redux store inside the context provider because at this point the redux store hasn't been initialized.

So the runTour context is set to a number (0 or 1) first, and if it is set to a zero or one, that's how we can tell we need to look at the redux store once it's been initialized which we do here:

`useEffect(() => {
    if (runTour === 1) {
      setRunTour(!getDefaultCheckboxValue())
    } else if (runTour === 0) {
      setRunTour(false)
    }
  }, [runTour, setRunTour])

  const shouldRunTour = () => runTour`
  
If we just return true/false from the `useContext`, we have no way to tell if that value came from the initial state. Because there's a situation where a logged in user has nothing in localstorage, so the `runTour` context defaults to true, but if they have the tour set to `dontshow` in their preferences, we run into issues with not knowing we need to update the tour context again. I can explain this more in detail in the PR walk through but that's the reason I went with this approach.